google/cloud/forseti/common/gcp_type/folder.py

```diff
@@ -56,3 +58,29 @@ def __init__(
             lifecycle_state=lifecycle_state)
         self.full_name = full_name
         self.data = data
+
+    @classmethod
+    def from_json(cls, parent, json_string):
+        """Creates a Folder from a JSON string.
+
+        Args:
+            parent (Resource): resource this folder belongs to.
+            json_string (str): JSON string of a folder GCP resource.
+
+        Returns:
+            Folder: folder resource.
+        """
+        folder_dict = json.loads(json_string)
+        name = folder_dict['name']
+        folder_id = name.split('/')[-1]
+        full_name = '{}folder/{}/'.format(parent.full_name, folder_id)
+        lifecycle = folder_dict.get('lifecycleState',
```

Comment:
Depending on the inventory, a folder may not have a parent, please do a check like https://github.com/GoogleCloudPlatform/forseti-security/blob/dev/google/cloud/forseti/common/gcp_type/project.py#L86.

And add an appropriate test.